arviz/data/inference_data.py

```diff
@@ -98,6 +98,25 @@ def __add__(self, other):
         """Concatenate two InferenceData objects."""
         return concat(self, other, copy=True, inplace=False)
 
+    def sel(self, **kwargs):
```

Comment:
what happens if the group is not in one of the groups/datasets?

Reply:
er, the selection, rather (will having `sampler_statistics` break this?)

Reply:
There is a check for every pair _key_-_group_ to ensure that the key is one of the dimensions in the group. Thus, no error is risen when performing the burn in on the whole inference data even though the observed_data group has no draw dimension. Is this what you mean?

Reply:
actually, what do you think about

```python
for group in self._groups:
    dataset = getattr(self, group)
    valid_keys = set(kwargs.keys()).intersection(dataset.dims)
    dataset = dataset.sel(**{key: kwargs[key] for key in valid_keys})
    setattr(self, group, dataset)
```

I think this is significantly faster for certain use cases.

Reply:
Definitely much better. I wasn't really sure if this would be useful to someone else other than me. I can add copy argument too.